samples/snippets/quickstart.py

```diff
@@ -0,0 +1,246 @@
+"""
+
+import argparse
+
+# [START compute_instances_list]
+# [START compute_instances_list_all]
+# [START compute_instances_create]
+# [START compute_instances_delete]
+# [START compute_instances_operation_check]
+import typing
+
+import google.cloud.compute_v1 as gce
+
+# [END compute_instances_operation_check]
+# [END compute_instances_delete]
+# [END compute_instances_create]
+# [END compute_instances_list_all]
+# [END compute_instances_list]
+
+
+# [START compute_instances_list]
+def list_instances(project: str, zone: str) -> typing.Iterable[gce.Instance]:
+    """
+    Gets a list of instances created in given project in given zone.
+    Returns an iterable collection of Instance objects.
+
+    Args:
+        project: Name of the project you want to use.
+        zone: Name of the zone you want to check, for example: us-west3-b
+
+    Returns:
+        An iterable collection of Instance objects.
+    """
+    instance_client = gce.InstancesClient()
+    instance_list = instance_client.list(project=project, zone=zone)
+    return instance_list
+
+
+# [END compute_instances_list]
+
+
+# [START compute_instances_list_all]
+def list_all_instances(project: str) -> typing.Dict[str, typing.Iterable[gce.Instance]]:
+    """
+    Returns a dictionary of all instances present in a project, grouped by their zone.
+
+    Args:
+        project: Name of the project you want to use.
+
+    Returns:
+        A dictionary with zone names as keys (in form of "zones/{zone_name}") and
+        iterable collections of Instance objects as values.
+    """
+    instance_client = gce.InstancesClient()
+    agg_list = instance_client.aggregated_list(project=project)
+    all_instances = {}
+    for zone, response in agg_list:
+        if response.instances:
+            all_instances[zone] = response.instances
+    return all_instances
+
+
+# [END compute_instances_list_all]
+
+
+# [START compute_instances_create]
+def create_instance(
+    project: str, zone: str, machine_type: str, machine_name: str, source_image: str
+) -> gce.Instance:
+    """
+    Sends an instance creation request to GCP and waits for it to complete.
+
+    Args:
+        project: Name of the project you want to use.
+        zone: Name of the zone you want to use, for example: us-west3-b
+        machine_type: Machine type you want to create in following format:
+            "zones/{zone}/machineTypes/{type_name}". For example:
+            "zones/europe-west3-c/machineTypes/f1-micro"
+        machine_name: Name of the new machine.
+        source_image: Path the the disk image you want to use for your boot
+            disk. This can be one of the public images
+            (e.g. "projects/debian-cloud/global/images/family/debian-10")
+            or a private image you have access to.
+
+    Returns:
+        Instance object.
+    """
+    instance_client = gce.InstancesClient()
+
+    # Every machine requires at least one persistent disk
+    disk = gce.AttachedDisk()
```

Comment:
(No action needed). As an FYI the attributes can also be set when the objects are created. e.g. `disk = gce.AttachedDisk(boot=True, ...)`.